### ISSUE TITLE
Automatic terminator refreshing

### ISSUE THREAD

**Issue by linkfanel** [joergdietrich/Leaflet.Terminator#16]
Calling `setTime()` refreshes the terminator to the current time. It would be nice if this could work out of the box with no additional client code.

There is always the `setInterval()` trick, but it's not necessarily a good design to add background timers that will suck out CPU cycles even when the page is inactive.

I've migrated from Google Maps and the day/night overlay at https://github.com/marmat/google-maps-api-addons This implementation refreshes the overlay automatically to the current time, upon interaction with the map - although it might be more of an incidental result due to the use of the built-in google.maps.OverlayView class. I felt pretty good about that behavior.

To replicate this, I use a snippet like:

```javascript
var terminator = L.terminator();
map.addEventListener('zoomstart movestart popupopen', function(e) {
        terminator.setTime();
});
terminator.addTo(map);
```

It's also possible to trigger refreshing more thoroughly on additional map interaction events, like for example 'baselayerchange' or the somewhat enigmatic 'viewreset' (https://leafletjs.com/reference-1.6.0.html#map-event).

It would be nice if either or both of these automatic refreshing mechanisms could be available and provided by the library, if not enabled by default - maybe also controlled by an option; or at least documented further by putting code samples in the README.

**Comment by joergdietrich** [comment]
Many thanks for your input and you suggestion. I agree that using CPU to update inactive pages is a design choice that should be made carefully and the current README doesn't spell this out. At this point I'm not sure I'd add code to the library to provide auto-update mechanisms. I'd probably prefer to keep this on the user side. If you wanted to amend the documentation accordingly, I'd be very happy to include your changes.

**Comment by joergdietrich** [comment]
Closed via #18.
